doc/nrf/ug_bootloader_external_flash.rst

```diff
@@ -10,60 +10,38 @@ This requires a driver for the external flash memory that supports the following
 * Writing data from the internal flash memory to the external flash memory.
 
 See :ref:`pm_external_flash` for general information about how to set up partitions in the external flash memory using the Partition Manager.
+Two more configurations must be set in order for the secondary partition to be stored in the external flash.
 
-.. note::
-
-   Currently, only MCUboot supports external flash memory partitions for image slots.
-
-To configure a bootloader to use the external flash memory for the secondary slot, edit the configuration file of the bootloader child image to specify the external flash memory region.
-For MCUboot, this configuration is :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml`.
-
-In this file, you must define the region under the section ``mcuboot_secondary``:
-
-.. code-block:: yaml
+* :kconfig:`CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY` - place the secondary partition of MCUboot in external flash.
+* :kconfig:`CONFIG_PM_PARTITION_SIZE_MCUBOOT_SECONDARY` - specifies the size of the secondary partition.
 
-   mcuboot_secondary:
-       region: external_flash
-       size: CONFIG_PM_EXTERNAL_FLASH_SIZE
+The value of :kconfig:`CONFIG_PM_PARTITION_SIZE_MCUBOOT_SECONDARY` requires manual configuration.
+This value must match the value of the automatically deduced size of the primary partition.
+A CMake warning is given when the values do not match, so the correct value can be found there.
 
-The nRF52840 DK comes with an external flash memory that can be used for the secondary slot and be accessed using the QSPI NOR flash memory driver.
-The following steps show how to configure an application for the nRF52840 DK:
+.. note::
 
-1. Add the following configuration options to the ``prj.conf`` file associated with the MCUboot child image:
+   The value of :kconfig:`CONFIG_PM_EXTERNAL_FLASH`, :kconfig:`CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY` and :kconfig:`CONFIG_PM_PARTITION_SIZE_MCUBOOT_SECONDARY` must be set for both the parent image and the MCUboot child image.
```

Comment:
This will no longer be the case when we get the improvement in partition manager which allows sharing the size across regions. Then we can simply forward the value of `CONFIG_PM_EXTERNAL_FLASH` and `CONFIG_PM_EXTERNAL_FLASH_SECONDARY` through overlays, and no manual synchronization of the options will be necessary.

Reply:
Is this improvement planned before the merge of this PR? If that's not the case, I would still leave this info there and update it at a later stage.

Reply:
We'll leave it then. We don't know exactly when the fix will arrive.